src/processor/operator/transaction.cpp

```diff
@@ -53,14 +57,17 @@ void Transaction::validateActiveTransaction(const TransactionContext& context) c
                 "connections.");
         }
     } break;
-    case TransactionAction::COMMIT: {
+    case TransactionAction::COMMIT:
+    case TransactionAction::ROLLBACK: {
         if (!context.hasActiveTransaction()) {
-            throw TransactionManagerException("No active transaction for COMMIT.");
+            throw TransactionManagerException(stringFormat("No active transaction for {}.",
+                TransactionActionUtils::toString(transactionAction)));
         }
     } break;
-    case TransactionAction::ROLLBACK: {
-        if (!context.hasActiveTransaction()) {
-            throw TransactionManagerException("No active transaction for ROLLBACK.");
+    case TransactionAction::CHECKPOINT: {
+        if (context.hasActiveTransaction()) {
+            throw TransactionManagerException(stringFormat("Found active transaction for {}.",
```

Comment:
Would it be better to say "Checkpoint is not allowed in an active transaction` here?